File: packages/flutter-dart/lib/appkit_auth/models/auth_error.dart
/// Auth errors
sealed class AuthError implements Exception {
  const AuthError(this.message);
  final String message;

  @override
  String toString() => 'AuthError: $message';
}

class InvalidURLError extends AuthError {
  const InvalidURLError() : super('Invalid URL');
}

class CancelledError extends AuthError {
  const CancelledError() : super('Authentication cancelled');
}

class InvalidCredentialsError extends AuthError {
  const InvalidCredentialsError() : super('Invalid credentials');
}

class TokenExpiredError extends AuthError {
  const TokenExpiredError() : super('Token expired');
}

class NotConfiguredError extends AuthError {
  const NotConfiguredError() : super('Auth not configured');
}

class NetworkAuthError extends AuthError {
  const NetworkAuthError(String detail) : super('Network error: $detail');
}

class NotImplementedError extends AuthError {
  const NotImplementedError(String feature) : super('Not implemented: $feature');
}

File: packages/android-kotlin/src/main/kotlin/com/cinacoin/appkit/auth/model/AuthError.kt
package com.cinacoin.appkit.auth.model

sealed class AuthError(message: String) : Exception(message) {
    object InvalidURL : AuthError("Invalid URL")
    object Cancelled : AuthError("Authentication cancelled")
    object InvalidCredentials : AuthError("Invalid credentials")
    object TokenExpired : AuthError("Token expired")
    object NotConfigured : AuthError("Auth not configured. Call configure() first.")
    data class NetworkError(val detail: String) : AuthError("Network error: $detail")
    data class NotImplemented(val feature: String) : AuthError("Not implemented: $feature")
}
